systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,112 @@
+        resourceManager.createResource(extensionContext, BundleResource.clusterOperator(NAMESPACE, envVars).build());
+    }
+
+    /**
+     * Control Plane Listener
+     * https://github.com/strimzi/proposals/blob/main/025-control-plain-listener.md
+     */
+    @IsolatedTest("Feature Gates test for ControlPlainListener")
+    @Tag(INTERNAL_CLIENTS_USED)
+    public void testControlPlaneListenerFeatureGate(ExtensionContext extensionContext) throws InterruptedException {
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String producerName = "producer-test";
+        final String consumerName = "consumer-test";
+        final String topicName = "topic-fg-cpl";
+        int sendingTime = 300;
+        List<EnvVar> testEnvVars = new ArrayList<>();
+        int kafkaReplicas = 2;
+
+        testEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+ControlPlaneListener", null));
+        deployClusterOperatorWithEnvVars(extensionContext, testEnvVars);
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, kafkaReplicas).build());
+
+        LOGGER.info("Check for presence of ContainerPort 9090/tcp (tcp-ctrlplane) in first Kafka pod.");
+        final Pod kafkaPod = PodUtils.getPodsByPrefixInNameWithDynamicWait(NAMESPACE, clusterName + "-kafka-").get(0);
+        ContainerPort expectedControlPlaneContainerPort = new ContainerPort(9090, null, null, "tcp-ctrlplane", "TCP");
+        List<ContainerPort> kafkaPodPorts = kafkaPod.getSpec().getContainers().get(0).getPorts();
+        assertTrue(kafkaPodPorts.contains(expectedControlPlaneContainerPort));
+
+        LOGGER.info("Try to send some messages to Kafka over the next {} seconds.", sendingTime);
+        // use example clients for jobs - sending messages for 3-5mins
+        KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBasicExampleClients.Builder()
+                .withProducerName(producerName)
+                .withConsumerName(consumerName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(topicName)
+                .withMessageCount(sendingTime)
+                .withDelayMs(1000)
+                .withNamespaceName(NAMESPACE)
+                .build();
+
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi().build());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi().build());
+
+        LOGGER.info("Remove kafka-broker service and wait for it's recreation (should be done within 30secs)");
+        KubeClusterResource.kubeClient().deleteService(KafkaResources.brokersServiceName(clusterName));
+        ServiceUtils.waitForServiceDeletion(KafkaResources.brokersServiceName(clusterName), NAMESPACE);
+        ServiceUtils.waitForServiceCreation(KafkaResources.brokersServiceName(clusterName), NAMESPACE);
```

Comment:
I would remove this. I do not think it adds any value in this test.

Reply:
removed